### PR TITLE
Replace 'request' with 'axios'

### LINKED ISSUE #213: Replace 'request' library
It seems the only place the `request` library is used is in `lib/Tunnel.js`, where it's only used for a simple task.
The `request` library is 52 packages, which take up around 4.22 MB, which seems unjustified for the small one-time usage that it gets included for. Would a PR be welcome to replace it with a lightweight alternative, such as [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch) or [simple-get](https://www.npmjs.com/package/simple-get) or something similar?

### PR DESCRIPTION
**Pull request by cktang88**
Fixes #213

### CODE REVIEW

File: lib/Tunnel.js [hunk 1]
```diff
@@ -26,8 +26,8 @@ Tunnel.prototype._init = function(cb) {
     var opt = self._opt;
 
     var params = {
-        path: '/',
-        json: true
+        //path: '/',
```
Comment (defunctzombie) [inline lib/Tunnel.js]: No need to keep dead unused code/options around.

File: lib/Tunnel.js [hunk 2]
```diff
@@ -39,34 +39,32 @@ Tunnel.prototype._init = function(cb) {
     var assigned_domain = opt.subdomain;
 
     // where to quest
-    params.uri = base_uri + ((assigned_domain) ? assigned_domain : '?new');
+    uri = base_uri + ((assigned_domain) ? assigned_domain : '?new');
```
Comment (defunctzombie) [inline lib/Tunnel.js]: missing `const/let/var` so this leaks

File: lib/Tunnel.js [hunk 3]
```diff
@@ -39,34 +39,32 @@ Tunnel.prototype._init = function(cb) {
...
-                console.log('tunnel server offline: ' + err.message + ', retry 1s');
-                return setTimeout(get_url, 1000);
-            }
-
-            if (res.statusCode !== 200) {
-                var err = new Error((body && body.message) || 'localtunnel server returned an error, please try again');
+        axios.get(uri, params)
+        .then(function(res){
+            var body = res.data;
+            if (res.status !== 200) {
```
Comment (defunctzombie) [inline lib/Tunnel.js]: What does axios consider a successful response vs an error? I believe there is a way to have axios consider all non-200 responses an error.

File: package.json [hunk 4]
```diff
@@ -9,10 +9,10 @@
     "url": "git://github.com/localtunnel/localtunnel.git"
   },
   "dependencies": {
-    "request": "2.81.0",
-    "yargs": "3.29.0",
+    "axios": "^0.17.1",
```
Comment (defunctzombie) [inline package.json]: Please follow existing dependency convention and remove the `^`. When contributing to projects always try to make your changes look like the nearby code/configuration style.

Comment (cktang88) [reply package.json]: fixed via 80198e1

File: lib/Tunnel.js (hunk 1 above)
Comment (cktang88) [reply lib/Tunnel.js]: fixed via d06a053

File: lib/Tunnel.js (hunk 3 above)
Comment (cktang88) [reply lib/Tunnel.js]: fixed via bbdbfd6

File: lib/Tunnel.js (hunk 2 above)
Comment (cktang88) [reply lib/Tunnel.js]: fixed via bbdbfd9

File: lib/Tunnel.js (hunk 3 above)
Comment (cktang88) [reply lib/Tunnel.js]: From [](https://github.com/axios/axios/blob/master/lib/adapters/http.js), axios considers various error conditions such as network timeout, improper request format, low-level network errors, etc. Everything else is technically considered by axios as a successful response.
In the [config](https://github.com/axios/axios#request-config) object, there's a `validateStatus` option that allows us to define conditions for accepting or rejecting the promise, but that wouldn't throw an error.
The code above currently calls the callback with the error for non-200 successful responses, and catches axios-defined errors using the catch() block further down.

**Pull request merged**